examples/edit_minidump.rs

```diff
@@ -0,0 +1,209 @@
+
+    // Collect the descriptors for mutating memory regions.
+    let mem_list: minidump::MinidumpMemoryList = dump
+        .get_stream()
+        .map_err(|e| e.compat())
+        .context("Failed to parse memory regions from minidump")?;
+    let mem_descriptors: Vec<minidump::format::MINIDUMP_MEMORY_DESCRIPTOR> =
+        mem_list.iter().map(|mem| mem.desc).collect();
+
+    // Collect the descriptors for mutating linux raw streams.
+    let mut raw_descriptors: Vec<minidump::format::MINIDUMP_LOCATION_DESCRIPTOR> = Vec::new();
+    let raw_stream_types = [
+        minidump::format::MINIDUMP_STREAM_TYPE::LinuxEnviron,
+        minidump::format::MINIDUMP_STREAM_TYPE::LinuxCmdLine,
+    ];
+    for stream_type in &raw_stream_types {
+        match dump.get_raw_stream(*stream_type) {
+            Ok(stream) => {
+                let stream_p = stream.as_ptr() as usize;
+                let data_p = data.as_ptr() as usize;
+                let offset = stream_p - data_p;
+                raw_descriptors.push(minidump::format::MINIDUMP_LOCATION_DESCRIPTOR {
+                    rva: offset as u32,
+                    data_size: stream.len() as u32,
+                });
+            }
+            Err(minidump::Error::StreamNotFound) => (),
+            Err(e) => return Err(e.compat()).context("Unexpected error getting raw stream"),
+        }
+    }
+
+    // Collect the descriptors to mutate the filenames in referenced modules.
+    let mod_list: minidump::MinidumpModuleList = dump
+        .get_stream()
+        .map_err(|e| e.compat())
+        .context("Failed to parse modules from minidump")?;
+    let mut file_name_descriptors: Vec<minidump::format::MINIDUMP_LOCATION_DESCRIPTOR> = Vec::new();
+    for module in mod_list.iter() {
+        let name_size: u32 = data.pread_with(module.raw.module_name_rva as usize, dump.endian)?;
+        file_name_descriptors.push(minidump::format::MINIDUMP_LOCATION_DESCRIPTOR {
+            rva: module.raw.module_name_rva + std::mem::size_of::<u32>() as u32,
+            data_size: name_size,
+        });
+        // TODO: add debug names
+        // TODO: deduplicate descriptors.  E.g. a code and debug name could both point to
+        // the same string.
+    }
+
+    // Time to modify things!
+
+    let mut changed = false;
+
+    for mem_desc in mem_descriptors {
+        let bytes_type = if stack_rvas.contains(&mem_desc.memory.rva) {
+            AttachmentBytesType::MinidumpStack
+        } else {
+            AttachmentBytesType::MinidumpHeap
+        };
+        let dest = data.desc_mut_slice(&mem_desc.memory);
+        assert!(dest.len() > 0);
+
+        changed |= pii_processor.scrub_attachment_bytes(&filename, dest, bytes_type);
+    }
+
+    for raw_desc in raw_descriptors {
+        changed |= pii_processor.scrub_attachment_bytes(
+            &filename,
+            data.desc_mut_slice(&raw_desc),
+            AttachmentBytesType::PlainAttachment,
+        );
```

Comment:
@untitaker this is extremely untested, it just compiles.  The format of these is just a bunch of bytes, which are "unix string" (aka any corrupted encoding possible) separated by NULL bytes.

Reply:
I wonder whether we can get the locale settings out of the minidump to figure out how to decode paths, env, and whatever this is.

Reply:
Yes, locale can normally be parsed out of the environ and used to parse out the encoding (my past self seems to think https://hg.sr.ht/~flub/syskit/browse/syskit/_utils.py?rev=tip#L109 is the way to get this, there's probably rust libs though).

Doing this gets back to how to reverse an encoding to modify the minidump though.